File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def fun(n):
            c=0
            while n:
                r=n%10
                n//=10
                c+=r**2
            return c
        while (n//10 !=0):
            n=fun(n)
        if (n==1 or n==7):return True
        return False

File: 202-happy-number/happy-number.cpp
class Solution {
public:
long long fun(int n){
    long long c=0;
    while (n){
        int r=n%10;
        n/=10;
        c+=r*r;
    }
    return c;

}
    bool isHappy(int n) {

        while (n/10 !=0){
            n=fun(n);
        }
        return (n==1 or n==7);
    }
};